include/deal.II/fe/fe.h

```diff
@@ -3345,6 +3351,16 @@ FiniteElement<dim, spacedim>::is_primitive(const unsigned int i) const
 
 
 
+template <int dim, int spacedim>
+inline const Table<2, bool> &
+FiniteElement<dim, spacedim>::get_local_dof_sparsity_pattern() const
+{
+  static Table<2, bool> return_value;
```

Comment:
Do we need a static variable here? Is it necessary to have this function in the header file?

Reply:
How do you return a const ref without storing an object somewhere? One could store a member in every finite element but I don't see how that is better.

Reply:
The question is if we would be fine to just return a table by value instead. All uses of this type get set up outside the loops.

Reply:
It would be nice to be consistent with the rest of the code in `FiniteElement`.  Part of the functions store the returned variables inside of the class others return take a reference as input.

> The question is if we would be fine to just return a table by value instead. All uses of this type get set up outside the loops.

This would be also an option. @bangerth did a lot of work for modernizing code where we replaced input references with return by value.